aggregator/operator_registrations_inmemory.go

```diff
@@ -0,0 +1,190 @@
+		// before the block at which we started the ws subscription above
+		wg.Done()
+
+		for {
+			select {
+			case <-ctx.Done():
+				ors.logger.Infof("OperatorRegistrationsServiceInMemory: Context cancelled, exiting")
+				return
+
+			case err := <-newPubkeyRegistrationSub.Err():
+				ors.logger.Error("Error in websocket subscription for new pubkey registration events. Attempting to reconnect...", "err", err, "service", "OperatorRegistrationsServiceInMemory")
+				newPubkeyRegistrationSub.Unsubscribe()
+				newPubkeyRegistrationC, newPubkeyRegistrationSub, err = ors.avsRegistrySubscriber.SubscribeToNewPubkeyRegistrations()
+				if err != nil {
+					ors.logger.Error("Error opening websocket subscription for new pubkey registrations", "err", err, "service", "OperatorRegistrationsServiceInMemory")
+					// see the warning above the struct definition to understand why we panic here
+					panic(err)
+				}
+
+			case newPubkeyRegistrationEvent := <-newPubkeyRegistrationC:
+				pubkeys := types.OperatorPubkeys{
+					G1Pubkey: bls.NewG1Point(newPubkeyRegistrationEvent.PubkeyG1.X, newPubkeyRegistrationEvent.PubkeyG1.Y),
+					G2Pubkey: bls.NewG2Point(newPubkeyRegistrationEvent.PubkeyG2.X, newPubkeyRegistrationEvent.PubkeyG2.Y),
+				}
+				operatorId := types.OperatorIdFromPubkey(pubkeys.G1Pubkey)
+				operatorAddr := newPubkeyRegistrationEvent.Operator
+
+				pubkeyByAddrDict[operatorAddr] = pubkeys
+				pubkeyByIdDict[operatorId] = pubkeys
+
+				ors.logger.Debug("Added operator pubkeys to pubkey dict",
+					"service", "OperatorRegistrationsServiceInMemory",
+					"block", newPubkeyRegistrationEvent.Raw.BlockNumber,
+					"operatorAddr", operatorAddr,
+					"operatorId", operatorId,
+					"G1pubkey", pubkeyByAddrDict[operatorAddr].G1Pubkey,
+					"G2pubkey", pubkeyByAddrDict[operatorAddr].G2Pubkey,
+				)
+
+			// Receive a queryByAddr from GetOperatorPubkeys
+			case operatorPubkeyQuery := <-queryByAddrC:
+				pubkeys, ok := pubkeyByAddrDict[operatorPubkeyQuery.operatorAddr]
+				operatorPubkeyQuery.respC <- resp{pubkeys, ok}
+
+			// Receive a queryById from GetOperatorPubkeysById
+			case operatorPubkeyQuery := <-queryByIdC:
+				pubkeys, ok := pubkeyByIdDict[operatorPubkeyQuery.operatorId]
+				operatorPubkeyQuery.respC <- resp{pubkeys, ok}
+			}
+		}
+	}()
+}
+
+func (ors *OperatorRegistrationsServiceInMemory) queryPastRegisteredOperatorEventsAndFillDb(ctx context.Context, pubkeyByAddrDict map[common.Address]types.OperatorPubkeys, pubkeyByIdDict map[types.OperatorId]types.OperatorPubkeys) {
+	// Querying with nil startBlock and stopBlock will return all events. It doesn't matter if we queryByAddr some events that we will receive again in the websocket,
+	// since we will just overwrite the pubkey dict with the same values.
+	alreadyRegisteredOperatorAddrs, alreadyRegisteredOperatorPubkeys, err := ors.avsRegistryReader.QueryExistingRegisteredOperatorPubKeys(ctx, nil, nil)
```

Comment:
Shouldn't we chunk blocks here? i.e. 10000 blocks / call.

Reply:
Another approach would be going with nil, nil first and chunks only if errored

Reply:
> Shouldn't we chunk blocks here? i.e. 10000 blocks / call.

This is how its done

Reply:
Btw, maybe we can add our custom impl to this as well. Calling in chunks should take way more time than an unbound block range call, so I'd actually prefer it (and should save us quite some time) if we tried unbound first before moving to chunks.

Reply:
As I remember that was a frequent issue, so I'm not sure we shall be concerned with this.

Do you remember request error after exceeding some time of execution? In that case it would be a problematic, since we would be blocked by unbound request for a while and then receive error that it "exceeded time limit and doesn't work anymore"

Reply:
No, at least the aggregator should use a node that supports that - currently we are. I'd say even not knowing the RPC it's still worth it. Making calls in chunks overall increases the call time considerably, it's just a matter of following block range limits of RPCs.

Reply:
Would be good to merge it in the current state and add required logic in separate PR